Assets/ModScripts/Submodules/UnscrewMaze.cs

```diff
@@ -0,0 +1,145 @@
+            movementNum = Arrow;
+        else
+        {
+            int[] movementIndices = { 2, 3, 1, 0 };
+            movementNum = Array.IndexOf(movementIndices, Info.Arrows[Arrow]);
+        }
+        if (!maze[curPos].Contains(movementNum.ToString()))
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! You hit a wall by moving {1} at the coordinates ({2}, {3}).", ModuleID, Info.ArrowDirections[movementNum].ToLower(), Math.Floor(curPos / 6f) + 1, (curPos % 6) + 1);
+            Module.CauseStrike();
+            return;
+        }
+        switch (movementNum)
+        {
+            case 0:
+                curPos -= 6;
+                break;
+            case 1:
+                curPos += 1;
+                break;
+            case 2:
+                curPos += 6;
+                break;
+            case 3:
+                curPos -= 1;
+                break;
+        }
+        UpdateMorse();
+    }
+
+    public override void OnBulbInteract(int Bulb)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.HandleBulbScrew(Bulb, BulbScrewedIn[Bulb], Info.BulbInfo[Bulb + 2]);
+
+        BulbScrewedIn[Bulb] = !BulbScrewedIn[Bulb];
+
+        Module.Audio.PlaySoundAtTransform(Module.BulbSounds[BulbScrewedIn[Bulb] ? 0 : 1].name, Module.transform);
+        Module.Bulbs[Bulb].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents() && !BulbScrewedIn[Bulb])
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} bulb was removed when the component selection was [{2}] instead of [{3}].", ModuleID, (Bulb + 1) == 1 ? "first" : "second", Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        if (bulbsSolved[Bulb])
+            return;
+
+        if (positions[Bulb + 1] != curPos)
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Bulb {1} incorrectly unscrewed at ({2}, {3}). Resetting maze position.", ModuleID, Bulb + 1, Math.Floor(curPos / 6f) + 1, (curPos % 6) + 1);
+            curPos = positions[0];
+            Module.CauseStrike();
+        }
+        bulbsSolved[Bulb] = true;
+
+        if (bulbsSolved[0] == true && bulbsSolved[1] == true)
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Both bulbs have been unscrewed. Module solved.", ModuleID);
+            Module.Solve();
+        }
+
+        return;
+    }
+
+    int[] Base36ToDec(string input)
+    {
+        string alpha = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        return input.Select(x => Array.IndexOf(alpha.ToArray(), x)).ToArray();
+    }
+
+    void UpdateMorse()
+    {
+        string alpha = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        Module.StopCoroutine(Module.MorseCodeAnimation);
+        string newMorse = alpha[curPos] + Info.Morse.Substring(1, 2);
```

Comment:
```suggestion
        Info.Morse = alpha[curPos] + Info.Morse.Substring(1, 2);
```
no need for a variable here.